source/extensions/original_ip_detection/custom_header/custom_header.cc

```diff
@@ -0,0 +1,51 @@
+#include "extensions/original_ip_detection/custom_header/custom_header.h"
+
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace OriginalIPDetection {
+namespace CustomHeader {
+
+CustomHeaderIPDetection::CustomHeaderIPDetection(
+    const envoy::extensions::original_ip_detection::custom_header::v3::CustomHeaderConfig& config)
+    : header_name_(config.header_name()),
+      allow_trusted_address_checks_(config.allow_extension_to_set_address_as_trusted()) {
+
+  if (config.has_reject_options()) {
+    Http::Code reject_code = Http::Code::Forbidden;
+
+    if (config.reject_options().has_status_on_error()) {
+      reject_code = static_cast<Http::Code>(config.reject_options().status_on_error().code());
+    }
+
+    reject_options_ = {reject_code, config.reject_options().body_on_error(),
```

Comment:
preference for default details if details_on_error is not present.
Honestly I doubt anyone would care to override the default so I'd be inclined to only have a default and we can add the override knob if someone wants it.